keep/common/management/commands/update_pidman_title.py

```diff
@@ -0,0 +1,247 @@
+                        task_name, \
+                        item.pid, \
+                        item.label, \
+                        pidman_label))
+                # when the names are the same
+                else:
+                    nochange_count += 1
+                    # nochange_log.write("[TIME]: %s, [PID]: %s, [FEDORA_LABEL]: %s, [PIDMAN_LABEL]: %s \n" % (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), item.pid, item.label, pidman_label) )
+                    nochange_log.write("%s, %s, %s, %s, %s\n" % \
+                        (time.strftime("%Y-%m-%d %H:%M:%S", \
+                        time.localtime()), \
+                        task_name, \
+                        item.pid, \
+                        item.label, \
+                        pidman_label))
+
+                # #TODO: remove this: write in the success log
+                # success_log.write("[TIME]: %s, [PID]: %s, [FEDORA_LABEL]: %s, [PIDMAN_LABEL]: %s \n" % (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), item.pid, item.label, pidman_label) )
+                # success_count += 1
+
+                # update progress on screen
+                msg = "Object %i of %i for %s" % (index, len(objects)-1, task_name)
+                sys.stdout.write(msg + chr(8) * len(msg))
+                sys.stdout.flush()
+                time.sleep(0.1)
+
+            # when any errors (exceptions) occur
+            except Exception, e:
+                fail_count += 1
+
+                # log the failure in a file
+                error_file_path = "%s/%s.log" % (self.error_path, item.noid)
+                error_log = open(error_file_path, 'w+')
+                error_log.write('[TIME]: %s, [CONTENT_MODEL]: %s, [PID]: %s\n %s \n' % \
+                    (time.strftime("%Y%m%d %H:%M:%S", time.localtime()), \
+                    task_name, \
+                    item.noid, \
+                    str(e)))
+                error_log.close()
+
+        # update progress to "DONE" when all items are iterated through
+        sys.stdout.write("DONE" + " "*len(msg)+"\n")
+        sys.stdout.flush()
+
+        # write statistics
+        summary_log.write("Total objects: %i \n" % total_count)
+        summary_log.write("Failed objects: %i \n" % fail_count)
+        summary_log.write("No change: %i | Change required: %i\n" % (nochange_count, change_count))
+        summary_log.close()
+
+    def get_pidman(self):
+        """Initialize a new Pidman client using the DjangoPidmanRestClient
+        wrapper. The credentials are pulled from the application settings.
+
+        Returns:
+            DjangoPidmanRestClient
+
+        """
+        # try to configure a pidman client to get pids.
+        try:
+            return DjangoPidmanRestClient()
+        except:
+            # if we're in dev mode then we can fall back on the fedora default
+            # pid allocator. in non-dev, though, we really need pidman
+            if getattr(settings, 'DEV_ENV', False):
```

Comment:
I think in this concept, the dev environment exception doesn't make sense - this script is useless without a pidman configuration.  So, instead I would catch the error and raise a `CommandError` that pidman needs to be configured for the script to run.